packages/bpk-docs/src/pages/DialogsPage/DialogsPage.js

```diff
@@ -0,0 +1,172 @@
+
+import DocsPageBuilder from './../../components/DocsPageBuilder';
+import Paragraph from './../../components/Paragraph';
+
+class DialogContainer extends Component {
+  constructor() {
+    super();
+
+    this.onOpen = this.onOpen.bind(this);
+    this.onClose = this.onClose.bind(this);
+    this.toggleWidth = this.toggleWidth.bind(this);
+
+    this.state = {
+      isOpen: false,
+      wide: false,
+    };
+  }
+
+  onOpen() {
+    this.setState({
+      isOpen: true,
+    });
+  }
+
+  onClose() {
+    this.setState({
+      isOpen: false,
+    });
+  }
+
+  toggleWidth() {
+    this.setState(state => ({
+      wide: !state.wide,
+    }));
+  }
+
+  render() {
+    const { buttonText, children, ...rest } = this.props;
+
+    return (
+      <div>
+        <BpkButton onClick={this.onOpen}>{buttonText}</BpkButton>
+        <BpkDialog id="my-dialog" isOpen={this.state.isOpen} onClose={this.onClose} wide={this.state.wide} {...rest}>
+          <div>{children}</div>
+          <BpkButtonLink onClick={this.onClose}>Close dialog</BpkButtonLink>
+        </BpkDialog>
+      </div>
+    );
+  }
+}
+
+DialogContainer.propTypes = {
+  buttonText: PropTypes.string.isRequired,
+  children: PropTypes.node.isRequired,
+};
+
+const components = [
+  {
+    id: 'default',
+    title: 'Default dialog',
+    blurb: [
+      <Paragraph>
+        The default has a title and close icon. Tapping the scrim or close icon will dismiss the dialog.
+      </Paragraph>,
+    ],
+    examples: [
+      <DialogContainer
+        title="Dialog title"
+        closeLabel="Close dialog"
+        buttonText="Open dialog"
+        getApplicationElement={() => document.getElementById('react-mount')}
+      >
+        <Paragraph>You can put anything you want in here.</Paragraph>
+      </DialogContainer>,
+    ],
+  },
+  {
+    id: 'non-dismissible',
+    title: 'Non dismissible',
+    blurb: [
+      <Paragraph>
+        In cases when you want the user to explicitly make or accept the choice, this configuration means the dialog
+        has no close icon, and tapping the scrim will NOT dismiss it. Instead, the user must interact with the dialogs
+        content in order to dismiss it.
+      </Paragraph>,
+    ],
+    examples: [
+      <DialogContainer
+        title="Dialog title"
+        closeLabel="Close dialog"
+        dismissible={false}
+        buttonText="Open dialog"
+        getApplicationElement={() => document.getElementById('react-mount')}
+      >
+        <Paragraph>You can put anything you want in here.</Paragraph>
+      </DialogContainer>,
+    ],
+  },
+  {
+    id: 'no-title',
+    title: 'No title',
+    blurb: [
+      <Paragraph>
+        You can configure the dialog to have no title, only a close icon.
+      </Paragraph>,
+    ],
+    examples: [
+      <DialogContainer
+        closeLabel="Close dialog"
+        buttonText="Open dialog"
+        getApplicationElement={() => document.getElementById('react-mount')}
```

Comment:
This should use `document.getElementById('application-container')` for the application element.